src/lib/comms/sol-coap.c

```diff
@@ -1018,22 +1037,50 @@ sol_coap_notify(struct sol_coap_server *server,
             sol_network_link_addr_to_str(&o->cliaddr, &addr);
             SOL_WRN("Failed to enqueue packet %p to %.*s", header,
                 SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&addr)));
-            goto error_pkt;
+            goto err_header;
         }
 
+        if (cb)
+            sol_coap_packet_unref(pkt);
         sol_coap_packet_unref(header);
     }
 
-    sol_coap_packet_unref(pkt);
+    if (!cb)
+        sol_coap_packet_unref(pkt);
+
     return r;
 
-err:
+err_header:
     sol_coap_packet_unref(header);
-error_pkt:
+err_pkt:
     sol_coap_packet_unref(pkt);
     return r;
 }
 
+SOL_API int
+sol_coap_notify(struct sol_coap_server *server,
+    struct sol_coap_resource *resource, struct sol_coap_packet *pkt)
+{
+    SOL_NULL_CHECK(pkt, -EINVAL);
+
+    return sol_coap_notify_full(server, resource, pkt, NULL, NULL);
+}
+
+SOL_API int
+sol_coap_notify_by_callback(struct sol_coap_server *server,
+    struct sol_coap_resource *resource,
+    int (*cb)(void *cb_data, struct sol_coap_server *server,
+    struct sol_coap_resource *resource, struct sol_network_link_addr *addr,
+    struct sol_coap_packet **pkt),
+    const void *cb_data)
+{
+    struct sol_coap_packet *pkt = NULL;
```

Comment:
I think it's more clear this way, since `sol_coap_notify` and `sol_coap_notify_full` already has a local with this name (third parameter).